### PR TITLE
Nft gallery multiple choice

### PR DESCRIPTION
**Pull request by rifeljm**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -32,6 +32,8 @@
     "eslint-comments/no-unused-disable": "error",
     // Allow single line disables (overrides the eslint-comments defaults)
     "eslint-comments/disable-enable-pair": "off",
+    // There are cases when Promise.All doesn't work for us (for example when we need to break a for loop - as in our case)
```
[the remark this answers is not in the document]
Comment (paninaro) [reply .eslintrc.json:35]: @rifeljm can you separate this out into its own PR? Thanks!

[the remark this answers is not in the document]
Comment (paninaro) [reply .eslintrc.json:35]: (or add the ignore-next line suggestion that @zsolt-dev made)

[the remark this answers is not in the document]
Comment (rifeljm) [reply .eslintrc.json:35]: @paninaro @zsolt-dev Ok, I understand. I will remove it from eslintrc.json

File: packages/core/src/components/CopyToClipboard/CopyToClipboard.tsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ import { useTimeout } from 'react-use-timeout';
 
 const StyledAssignmentIcon = styled(({ invertColor, ...rest }) => <AssignmentIcon {...rest} />)(
   ({ theme, invertColor }) => `
-  color: ${invertColor ? theme.palette.common.white : theme.palette.text.secondary};
+  color: ${invertColor ? theme.palette.common.white : '#757575'};
```
Comment (paninaro) [inline packages/core/src/components/CopyToClipboard/CopyToClipboard.tsx:12]: The #757575 color used for non-inverted mode doesn't look good in dark mode. Settings --> Profiles is an example where the non-inverted color is used:

image

File: packages/gui/src/components/nfts/gallery/NFTGallery.tsx [hunk 3]
```diff
@@ -117,70 +211,257 @@ export default function NFTGallery() {
...
+            <Search onChange={setSearch} placeholder={t`Search...`} />
+            <WalletReceiveAddressField variant="outlined" size="small" fullWidth />
+            <MultiSelectAndFilterWrapper className={inMultipleSelectionMode ? 'active' : ''}>
+              <MultiSelectIcon onClick={() => toggleMultipleSelection(!inMultipleSelectionMode)} />
+              <FilterIcon />
+            </MultiSelectAndFilterWrapper>
+          </Flex>
+
+          <Flex gap={2} alignItems="center" flexWrap="wrap" justifyContent="space-between" sx={{ padding: '10px 0' }}>
+            {t`Showing ${showCount()} of ${allowNFTsFiltered.length} items`}
```
Comment (paninaro) [inline packages/gui/src/components/nfts/gallery/NFTGallery.tsx:350]: If I switch to a wallet without any NFTs, the total count doesn't show properly. In the screenshot below, I switched from my "All NFTs" view to "Profile 1" (which doesn't have any NFTs).

image

File: packages/core/src/hooks/useHiddenList.ts [hunk 4]
```diff
@@ -10,7 +10,8 @@ export default function useHiddenList<Type>(
 ): [
   isHidden: (key: Type) => boolean,
   setIsHidden: (key: Type, newValue: (isHidden: boolean) => boolean | boolean) => void,
-  hidden: Type[]
+  hidden: Type[],
+  setIsNFTMultipleHide: (nftIds: string[]) => void
```
Comment (paninaro) [inline packages/core/src/hooks/useHiddenList.ts:14]:
```suggestion
  setIsNFTMultipleHide: (nftIds: string[], hide: boolean) => void
```

File: packages/gui/package.json [hunk 5]
```diff
@@ -150,6 +150,7 @@
     "bech32": "2.0.0",
     "bignumber.js": "9.0.2",
     "byte-size": "8.1.0",
+    "chokidar": "^3.5.3",
```
Comment (paninaro) [inline packages/gui/package.json:153]: Let's pin the version like we do with the others

```suggestion
    "chokidar": "3.5.3",
```

File: packages/gui/src/components/nfts/NFTCardLazy.tsx [hunk 6]
```diff
@@ -8,7 +8,7 @@ export type NFTCardLazyProps = NFTCardProps & {
   minHeight?: number;
 };
 
-export default function NFTCardLazy(props: NFTCardLazyProps) {
+export default React.memo((props: NFTCardLazyProps) => {
```
Comment (paninaro) [inline packages/gui/src/components/nfts/NFTCardLazy.tsx:11]: Why is this change necessary?

File: packages/gui/src/components/nfts/NFTContextualActions.tsx [hunk 7]
```diff
@@ -630,6 +667,13 @@ export default function NFTContextualActions(props: NFTContextualActionsProps) {
...
           },
         },
+        {
+          action: NFTViewOnExplorerContextualAction,
+          props: {
+            title: <Trans>View on SkyNFT</Trans>,
+            explorer: NFTExplorer.SkyNFT,
+          },
+        },
         {
```
Comment (paninaro) [inline packages/gui/src/components/nfts/NFTContextualActions.tsx:677]: Remove. SkyNFT no longer exists.

File: packages/gui/src/electron/main.tsx [hunk 8]
```diff
@@ -225,8 +246,8 @@ if (!handleSquirrelEvent()) {
       function getRemoteFileSize(urlLocal: string): Promise<number> {
         return new Promise((resolve, reject) => {
           axios({
-            method: 'get',
-            url: urlLocal,
+            method: 'HEAD',
+            url,
```
Comment (paninaro) [inline packages/gui/src/electron/main.tsx:250]: This breaks fetching metadata because it's referencing the wrong variable.

```suggestion
            url: urlLocal,
```

File: packages/gui/src/electron/main.tsx [hunk 9]
```diff
@@ -307,8 +328,7 @@ if (!handleSquirrelEvent()) {
           let totalLength = 0;
 
           try {
-            let fileSize: number;
-            fileSize = await getRemoteFileSize(rest.url);
+            const fileSize: number = await getRemoteFileSize(rest.url);
```
Comment (paninaro) [inline packages/gui/src/electron/main.tsx:331]: On line 302, rest.uri should be rest.url:

`          if (allRequests[rest.uri]) {`

should be 
`          if (allRequests[rest.url]) {`

File: packages/gui/src/components/nfts/NFTCard.tsx [hunk 10]
```diff
@@ -23,44 +20,90 @@ const StyledLoadingCardContent = styled(CardContent)`
...
+  height: 50px;
+  width: 100%;
+`;
+
 export type NFTCardProps = {
   nft: NFTInfo;
   canExpandDetails: boolean;
   availableActions: NFTContextualActionTypes;
   isOffer: boolean;
+  selctedItemAction?: () => void;
```
Comment (paninaro) [inline packages/gui/src/components/nfts/NFTCard.tsx:67]: Typo

```suggestion
  selectedItemAction?: () => void;
```

File: packages/gui/src/components/nfts/NFTCard.tsx [hunk 11]
```diff
@@ -23,44 +20,90 @@ const StyledLoadingCardContent = styled(CardContent)`
...
       navigate(`/dashboard/nfts/${nft.$nftId}`);
     }
   }
 
   return (
     <Flex flexDirection="column" flexGrow={1}>
-      <Card sx={{ borderRadius: '8px' }} variant="outlined">
-        {isLoading ? (
+      <CardWrapper sx={{ borderRadius: '8px' }} variant="outlined" className="card-wrapper">
+        {false ? (
```
Comment (paninaro) [inline packages/gui/src/components/nfts/NFTCard.tsx:94]: Should we remove this?

File: packages/gui/src/components/nfts/NFTCardLazy.tsx [hunk 12]
```diff
@@ -8,7 +8,7 @@ export type NFTCardLazyProps = NFTCardProps & {
   minHeight?: number;
 };
 
-export default function NFTCardLazy(props: NFTCardLazyProps) {
+export default function (props: NFTCardLazyProps) {
```
Comment (paninaro) [inline packages/gui/src/components/nfts/NFTCardLazy.tsx:11]:
```suggestion
export default function NFTCardLazy(props: NFTCardLazyProps) {
```

File: packages/gui/src/components/offers2/OfferBuilderValueSearch.tsx [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+        <span className="highlight">{part}</span>
+      ) : (
+        <span>{part}</span>
+      )
+    );
+}
+
+export default function OfferBuilderValueSearch(props: OfferBuilderValueSearchProps) {
+  const { value, onSelectNFT } = props;
+  const [walletId] = usePersistState<number | undefined>(undefined, 'nft-profile-dropdown');
```
Comment (paninaro) [inline packages/gui/src/components/offers2/OfferBuilderValueSearch.tsx:95]: The last selection in the NFT profile dropdown doesn't seem relevant to the search feature when creating an offer.

File: packages/gui/src/components/offers2/OfferBuilderValueSearch.tsx [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+  const { filteredNFTs, isLoading } = useFilteredNFTs({ walletId });
```
Comment (paninaro) [inline packages/gui/src/components/offers2/OfferBuilderValueSearch.tsx:96]: I'm not convinced this is the right approach. I think the search should be across all NFTs, not just the last grouping of NFTs that was filtered/searched in the gallery.

File: packages/gui/src/components/offers2/CreateOfferBuilder.tsx [hunk 15]
```diff
@@ -17,17 +17,20 @@ type createDefaultValuesParams = {
   walletType?: WalletType; // CAT or STANDARD_WALLET (XCH), indicates whether a token or CAT has a default entry
   assetId?: string; // Asset ID of the CAT
   nftId?: string; // NFT to include in the offer by default
+  nftIds?: string[]; // multiple NFT selection
   nftWalletId?: number; // If set, indicates that we are offering the NFT, otherwise we are requesting it
```
Comment (paninaro) [inline packages/gui/src/components/offers2/CreateOfferBuilder.tsx:21]: This is sort of a mess. Since we're now dealing with (possibly) multiple NFTs coming from different NFT wallets, we need to have a pair of NFT ID along with its wallet ID.

I'd recommend replacing these 3 props with a single `nftIdsAndWalletIds` prop that contains an array of `{ nftId: string, walletId: number }` objects.

Let's revisit this after merging this PR since it will be a bigger change.

File: packages/gui/src/components/nfts/gallery/NFTGallery.tsx (hunk 3 above)
Comment (paninaro) [reply packages/gui/src/components/nfts/gallery/NFTGallery.tsx:391]: Tracked as a new issue: https://github.com/Chia-Network/chia-blockchain-gui/issues/1250

File: packages/gui/src/components/offers2/OfferBuilderValueSearch.tsx (hunk 13 above)
Comment (paninaro) [reply packages/gui/src/components/offers2/OfferBuilderValueSearch.tsx:95]: Tracked as https://github.com/Chia-Network/chia-blockchain-gui/issues/1251

File: packages/gui/src/components/offers2/OfferBuilderValueSearch.tsx (hunk 14 above)
Comment (paninaro) [reply packages/gui/src/components/offers2/OfferBuilderValueSearch.tsx:96]: Tracked as https://github.com/Chia-Network/chia-blockchain-gui/issues/1251

File: packages/gui/src/components/offers2/CreateOfferBuilder.tsx (hunk 15 above)
Comment (paninaro) [reply packages/gui/src/components/offers2/CreateOfferBuilder.tsx:21]: Tracked as https://github.com/Chia-Network/chia-blockchain-gui/issues/1252

**Review by paninaro — APPROVED** [review]
Approved with a few issues to follow-up on
